plugins/studymaster-web/src/main/java/de/dfki/vsm/xtension/studymasterWeb/WebStudyMasterExecutor.java

```diff
@@ -0,0 +1,306 @@
+        mSceneflowGoVar = mConfig.getProperty(sGO_VAR, sGO_VAR_DEFAULT);
+
+        // Start the HTTP server
+        mHttpServer = Javalin.create(config -> {
+            config.addStaticFiles("/react-studymaster/build");
+            config.enforceSsl = true;
+        }).start(http_port);
+
+        // Set callbacks to manage WebSocket events
+        mHttpServer.ws("/ws", ws -> {
+            ws.onConnect(this::addWs);
+            ws.onMessage(ctx -> this.handleGUIMessage(ctx.message()));
+            ws.onClose(this::removeWs);
+            ws.onError(ctx -> mLogger.failure("WebSocket Error: " + ctx.error()));
+        });
+    }
+
+
+    @Override
+    public void unload() {
+        for (WsConnectContext ws: mWebsockets) {
+            if (ws.session.isOpen()) {
+                ws.session.close();
+            }
+        }
+        mWebsockets.clear();
+        mHttpServer.stop();
+    }
+
+
+    /** Invoked when a new websocket connection is opened (new web app is loaded).*/
+    private synchronized void addWs(WsConnectContext ws) {
+        mLogger.message("New WebSocket connection.");
+        this.mWebsockets.add(ws);
+
+        // Update the connection status
+        if (mProject.hasVariable(mGUIConnectedVar)) {
+            mProject.setVariable(mGUIConnectedVar, true);
+        }
+
+        // If a request is pending, send it to the new client
+        if(mLastRequestMessage != null) {
+            ws.send(mLastRequestMessage) ;
+        }
+
+    }
+
+
+    /** Invoked when a websocket connection is closed (web app is closed, or timeout?).*/
+    private synchronized void removeWs(WsCloseContext ctx) {
+        mLogger.message("Closed WebSocket connection.");
+        mWebsockets.remove(ctx);
+
+        // Set the GUIState variable to false if there are no more GUI connections.
+        if(mWebsockets.size()==0) {
+            if (mProject.hasVariable(mGUIConnectedVar)) {
+                mProject.setVariable(mGUIConnectedVar, false);
+            }
+
+        }
+    }
+
+
+
+    @Override
+    public void execute(AbstractActivity activity) {
+
+        if (activity instanceof SpeechActivity) {
+            SpeechActivity sa = (SpeechActivity) activity;
+            String text = sa.getTextOnly("$(").trim();
+            LinkedList<String> timemarks = sa.getTimeMarks("$(");
+
+            // If text is empty - assume activity has empty text but has marker activities registered
+            if (text.isEmpty()) {
+                for (String tm : timemarks) {
+                    mLogger.warning("Directly executing activity at timemark " + tm);
+                    mProject.getRunTimePlayer().getActivityScheduler().handle(tm);
+                }
+            }
+        } else {
+            final String action_name = activity.getName();
+            final LinkedList<ActionFeature> features = activity.getFeatures();
+
+            if (action_name.equalsIgnoreCase("stop")) {
```

Comment:
why ignore case? This only allows for sloppy code (opinion)

Reply:
I agree. I found it like that and repeated for further entries. Changed to "equals".